var/spack/repos/builtin/packages/xorgproto/package.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+from spack.package import *
+
+
+class Xorgproto(MesonPackage):
+    """X Window System unified protocol definitions replacing standalone protos"""
+
+    homepage = "https://gitlab.freedesktop.org/xorg/proto/xorgproto"
+    url = "https://gitlab.freedesktop.org/xorg/proto/xorgproto/-/archive/xorgproto-2024.1/xorgproto-xorgproto-2024.1.tar.bz2"
+
+    maintainers("teaguesterling")
+
+    license("OTHER", checked_by="teaguesterling")
+
+    version("2024.1", sha256="3959b2d17d86dd9d165dc24d26f372ca64f27127cd381739366ba8383a6cd51a")
```

Comment:
Iff switching to gz in xorg_mirror_path:
```suggestion
    version("2024.1", sha256="4f6b9b4faf91e5df8265b71843a91fc73dc895be6210c84117a996545df296ce")
```